src/PayID/pay-id-utils.ts

```diff
@@ -13,13 +11,38 @@ const payIdUtils = {
    * @returns A PayIdComponents object if the input was valid, otherwise undefined.
    */
   parsePayId(payId: string): PayIdComponents | undefined {
-    try {
-      utils.parsePayId(payId)
-      const [path, host] = utils.splitPayIdString(payId)
-      return new PayIdComponents(host, `/${path}`)
-    } catch {
+    if (!this.isASCII(payId)) {
+      return undefined
+    }
+
+    // Split on the last occurrence of '$'
+    const lastDollarIndex = payId.lastIndexOf('$')
+    if (lastDollarIndex === -1) {
+      return undefined
+    }
+    const path = payId.slice(0, lastDollarIndex)
+    const host = payId.slice(lastDollarIndex + 1)
+
+    // Validate the host and path have values.
+    if (host.length === 0 || path.length === 0) {
       return undefined
     }
+
+    return new PayIdComponents(host, `/${path}`)
+  },
+
+  /**
+   * Validate if the input is ASCII based text.
+   *
+   * Shamelessly taken from:
+   * https://stackoverflow.com/questions/14313183/javascript-regex-how-do-i-check-if-the-string-is-ascii-only.
+   *
+   * @param input - The input to check.
+   * @returns A boolean indicating the result.
+   */
+  isASCII(input: string): boolean {
```

Comment:
It's not clear to me that we need ASCII, since in most languages, requesting a unicode URL will automatically be transcribed into the underlying ASCII representation, so in that sense, unicode PayIDs are potentially valid.

Reply:
Done.